plugin.video.cinear/addon.py

```diff
@@ -0,0 +1,522 @@
+        context_menu.append((translation(30023), run_plugin('user_qualify', prod, params)))
+
+    # ordenar: no para capítulos ni para elementos de compilados
+    if 'url' in params:
+        context_menu.append((translation(30024), run_plugin('sort&url=%s' % quote(params['url']), prod, params)))
+
+    # agrega los menues contextuales
+    item.addContextMenuItems(context_menu, replaceItems=True)
+
+    if has_subprods:
+        url = '{0}?action={1}&pid={2}&source={3}&sid={4}'.format(
+                addon_url, 'list_subprods', PID, prod['id']['source'], prod['id']['sid'])
+        url += '&full=1' if 'compi' in prod['tags'] else ''
+        is_folder = True
+    else:
+        url = '{0}?action={1}&pid={2}&source={3}&sid={4}'.format(
+                addon_url, 'play', PID, prod['id']['source'], prod['id']['sid'])
+        item.setProperty('IsPlayable', 'true') # necesario solo si se usa xbmcplugin.setResolvedUrl()
+        item.addStreamInfo('video', {'codec': 'h264'})
+        item.addStreamInfo('audio', {'codec': 'aac', 'language' : 'es'})
+        is_folder = False
+
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=item, isFolder=is_folder)
+
+
+def toggle_misala(params):
+    path = f"{params['source']}/amisala/{PID}/{params['sid']}"
+    json_request(path)
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def user_qualify(params):
+    # mi calificación es
+    voto = xbmcgui.Dialog().select(translation(30025), ['1', '2', '3', '4', '5'])
+    if voto == -1: return
+    path = '{0}/votar/{1}/{2}?voto={3}'.format(params['source'], PID, params['sid'], voto + 1)
+    response = requests.post(make_url(path), headers=get_headers(addon.getSetting('token')))
+    if response.status_code == 200:
+        xbmc.executebuiltin("Container.Refresh")
+    else:
+        xbmcgui.Dialog().ok('Error ({0})'.format(response.status_code), response.content)
+        sys.exit(0)
+
+
+def sort(params):
+    # ordenar los ítems
+    order_names = [ord['nom'] for ord in META['order']]
+    order_ids = [ord['id'] for ord in META['order']]
+    index = xbmcgui.Dialog().select(translation(30026), order_names)
+    if index == -1: return
+    query = '{0}?action={1}&pid={2}&url={3}&orden={4}'.format(addon_url, 'list_prods', PID, quote(params['url']), order_ids[index])
+    xbmc.executebuiltin(f"Container.Update({query})")
+
+
+def search(params):
+    # título, director, actor...
+    texto = xbmcgui.Dialog().input(translation(30027))
+    if texto == '': return
+    params['url'] = 'search/' + quote(texto)
+    list_prods(params)
+
+
+def heartbeat(source, sid, currentTime, totalTime):
+    token = addon.getSetting('token')
+    url = make_url('{0}/vista/{1}/{2}/{3}/{4}/{5}'.format(source, PID, sid, int(totalTime), int(currentTime), token))
+    r = requests.get(url, headers=get_headers())
+    # xbmc.executebuiltin('Notification(%s, %s, %d, %s)'%(addon_name, 'Beat ' + str(currentTime/60), 1000, addon_icon))
+
+
+def monitor(source, sid):
```

Comment:
This is the only method the plaforms has to track a stream and provide a cross-platform way to resume a movie.
If it's problematic I can take it out completely but we will lose features and statistics.
I don´t know any other way to implement that. If you can provide an example I can see it later.

Reply:
My understanding is that plugins should only be tasked to resolve directories and videos, as this might stall some core parts otherwise. For me, this part would have to be executed as a script.

This is more of a "heads up" for the other reviewers, as I'm only helping out a bit :).

Reply:
It's fine as is, there's no other way to track and sync playback. The issue you are probably mentioning refers to the use of player.play() instead of resolvedurl which this plugin does not use :) thanks for the heads up